lib/bridge.js

```diff
@@ -972,57 +963,101 @@ function queueAlgorithm(event) {
+    constructor(opt, consumeFn) {
+        this.type = opt.type;
+        this._queues = {
+            // $identifier: {
+            //  events: [ {data: promise: } ],
+            //  consuming: true|false
+            // }
         };
+        this.consumeFn = consumeFn;
     }
-    this._queues[identifier].events.push({
-        data: data,
-        promise: promise
-    });
-};
 
-EventQueue.prototype.consume = function() {
-    var self = this;
-    Object.keys(this._queues).forEach(function(identifier) {
-        if (!self._queues[identifier].consuming) {
-            self._queues[identifier].consuming = true;
+    push(event, data, promise) {
+        var identifier = this.type === "per_room" ? event.room_id : "none";
+        if (!this._queues[identifier]) {
+            this._queues[identifier] = {
+                events: [],
+                consuming: false
+            };
+        }
+        this._queues[identifier].events.push({
+            data: data,
+            promise: promise
+        });
+    }
+
+    consume() {
+        var self = this;
+        Object.keys(this._queues).forEach(function(identifier) {
+            if (!self._queues[identifier].consuming) {
+                self._queues[identifier].consuming = true;
+                self._takeNext(identifier);
+            }
+        });
+    };
+
+    _takeNext(identifier) {
+        var self = this;
+        var events = this._queues[identifier].events;
+        if (events.length === 0) {
+            this._queues[identifier].consuming = false;
+            return;
+        }
+        var entry = events.shift();
+        entry.promise.done(function() {
+            self.consumeFn(null, entry.data);
+            self._takeNext(identifier);
+        }, function(e) {
+            self.consumeFn(e, null);
             self._takeNext(identifier);
+        });
+    };
+
+    static create(opts, consumeFn) {
+        const type = opts.type;
+        if (type == "single") {
+            return new EventQueueSingle(consumeFn);
         }
-    });
-};
+        if (type == "per_room") {
+            return new EventQueuePerRoom(consumeFn);
+        }
+        if (type == "none") {
+            return new EventQueueNone(consumeFn);
+        }
+        throw Error(`Invalid EventQueue type '${type}'.`);
+    }
+}
 
-EventQueue.prototype._takeNext = function(identifier) {
-    var self = this;
-    var events = this._queues[identifier].events;
-    if (events.length === 0) {
-        this._queues[identifier].consuming = false;
-        return;
+class EventQueueSingle extends EventQueue {
+    constructor(consumeFn) {
+        super({type: "single"}, consumeFn);
     }
-    var entry = events.shift();
-    entry.promise.done(function() {
-        self.consumeFn(null, entry.data);
-        self._takeNext(identifier);
-    }, function(e) {
-        self.consumeFn(e, null);
-        self._takeNext(identifier);
-    });
-};
+}
+
+class EventQueuePerRoom extends EventQueue {
```

Comment:
Thank you for using ES6 classes, it's quite refreshing :). Could the EventQueue stuff be taken to a seperate EventQueue.js file, as it makes for easier reviewing